optimlol_api/controllers/summonerController.js

```diff
@@ -148,7 +148,12 @@ module.exports = function() {
 				var championStats = results[promiseObject.STATS_INDEX].state === 'fulfilled' ? results[promiseObject.STATS_INDEX].value : null;
 				var recentHistoryStats = results[promiseObject.RECENT_STATS_INDEX].state === 'fulfilled' ? results[promiseObject.RECENT_STATS_INDEX].value : null;
 				var champions = results[promiseObject.CHAMPIONS_INDEX].state === 'fulfilled' ? results[promiseObject.CHAMPIONS_INDEX].value : null;
-				
+
+				summoner.dataQuality = "fresh";
+				if(championStats.quality === "stale" || recentHistoryStats.quality === "stale" || champions.quality === "stale") {
+					summoner.dataQuality = "stale";
+				}
+
```

Comment:
Ohhh shit...yea we might want to change that to pass the entire object so we can access the dataQuality variable. I just don't think it makes sense to be part of the data....also if data is null (no data in cache) then you can't set dataQuality of null in the dataProvider anyway.